db/migrate/20150103133015_move_prepaid_until_to_deployments.rb

```diff
@@ -0,0 +1,45 @@
+# The purpose of this migration is to move the 'prepaid_until' column from Atmosphere::Appliance to Atmosphere::Deployment
+# The reason for this is that an Appliance may have multiple deployments and new deployments may be added on the fly
+# while the Appliance is running - in such cases, it is no longer sufficient to store a global 'prepaid_until' timestamp
+# for the whole Appliance - rather, the billing system must iterate over the Appliance's deployments and bill each one
+# separately.
+class MovePrepaidUntilToDeployments < ActiveRecord::Migration
+  def up
+    add_column :atmosphere_deployments, :billing_state, :string, null:false, default:"prepaid"
+
+    # Time for some good old-fashioned SQL because the Rails ORM model is too constraining to support current timestamp as default datetime
+    execute "ALTER TABLE atmosphere_deployments ADD COLUMN prepaid_until TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP"
+
+    # Now we need to iterate over existing Appliances and rewrite the relevant data in order to maintain appliance info
+    Atmosphere::Appliance.find_each do |a|
+      a.deployments.each do |dep|
```

Comment:
n+1 queries here. You can [eager load](http://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations) deployments to avoid additional db queries.